README.md

```diff
@@ -19,141 +20,189 @@ When the reader has completed this Code Pattern, they will understand how to:
+* An instance of [Cloud Pak for Data](https://www.ibm.com/products/cloud-pak-for-data).
+* Watson OpenScale provisioned on Cloud Pak for Data.
+* [IBM Cloud Object Storage (COS)](https://www.ibm.com/cloud/object-storage)
 
 # Steps
 
 1. [Clone the repo](#1-clone-the-repo)
-2. [Create Watson services with IBM Cloud](#2-create-watson-services-with-ibm-cloud)
-3. [Create a notebook in IBM Watson Studio](#3-create-a-notebook-in-ibm-watson-studio) for use with a publicly addressed server OR
-   - Run the notebook locally for local testing only
-4. Perform either 4a to Deploy to IBM Cloud **or** 4b for local testing only:
-   - 4a. [Deploy to IBM Cloud](#4a-deploy-to-ibm-cloud)
-   - 4b. [Run the application server locally](#4b-run-the-application-server-locally)
-5. [Run the notebook in IBM Watson Studio](#5-run-the-notebook-in-ibm-watson-studio)
+1. [Create custom machine learning provider](#2-create-custom-machine-learning-provider)
+1. [Create a Watson OpenScale service](#3-create-a-watson-openscale-service)
+1. [Create COS bucket and get credentials](#4-create-cos-bucket-and-get-credentials)
+1. [Create a notebook on Cloud Pak for Data](#5-create-a-notebook-on-cloud-pak-for-data)
+1. [Run the notebook](#6-run-the-notebook)
 
-### 1. Clone the repo
+## 1. Clone the repo
 
 Clone the `monitor-custom-ml-engine-with-watson-openscale` locally. In a terminal, run:
 
 ```bash
 git clone https://github.com/IBM/monitor-custom-ml-engine-with-watson-openscale
 ```
 
-### 2. Create Watson services with IBM Cloud
+## 2. Create custom machine learning provider
 
-> Note: If you are using [Watson Studio]() for your notebook, services created must be in the same region, and space, as your Watson Studio service.
+Use your existing machine learning providor
 
-Create the following services:
+<details><summary>Setup on a VM</summary>
 
-* [Watson OpenScale](https://cloud.ibm.com/catalog/services/ai-openscale)
-  You will get the Watson OpenScale instance GUID when you run the notebook using the [IBM Cloud CLI](https://cloud.ibm.com/catalog/services/ai-openscale)
+### Custom Machine Learning Provider Setup for VM
 
-* [Databases for PostgreSQL DB](https://cloud.ibm.com/catalog/services/databases-for-postgresql)
+The code in flask.py can be used to start a gunicorn/flask application that can be hosted in a VM, such that it can be accessable from CPD system.
+This code does the following:
+* It wraps a Watson Machine Learning model that is deployed to a space.
+* The hosting application URL should contain the SPACE ID and the DEPLOYMENT ID. The app can be used to talk to the target WML model/deployment.
+* Having said that, this is only for this tutorial purpose, and you can define your Custom ML provider endpoint in any fashion you want, such that it wraps your own custom ML engine.
+* The scoring request and response payload should confirm to the schema as described here at: https://dataplatform.cloud.ibm.com/docs/content/wsj/model/wos-frameworks-custom.html
```

Comment:
"should confirm to the schema" - not sure what this means

Reply:
Should be "conform"